segfault in func ephemeralStorage of instancetype.go

### Version

<!-- helm ls -A --all -o json | jq '.[] | select(.name=="karpenter") | .app_version' -r -->
__Karpenter Version:__ v0.4.0
<!-- kubectl version | grep Server -->
__Kubernetes Version:__ v1.0.0


### Expected Behavior

While trying to run the instance type documentation generation program of the AWS provider with minor modifications like linking against the Azure provider, I expected to get the Azure instance types listed in a nice markdown file.

### Actual Behavior

Segfault. The nodeClass.Spec.OSDiskSizeGB pointer is zero in that function, at least if the provider GetInstanceTypes function is called with a nil nodepool. A nil nodepool is supposed to be a valid function argument and is used during the (AWS provider) documentation generation. The code dereferences nodeClass.Spec.OSDiskSizeGB pointer and it segfaults.

### Steps to Reproduce the Problem

Try calling the GetInstanceTypes function with a nil nodepool, like it is done in the documentation generator program. Example: https://github.com/aws/karpenter-provider-aws/blob/7b51ac1d90a2f90e7b882c508adacb91cd728acd/hack/docs/instancetypes_gen_docs.go#L107

### Resource Specs and Logs

This happens without any pods or Karpenter being run, in a separate program using the provider for listing instance types.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b2f602]

goroutine 1 [running]:
github.com/Azure/karpenter-provider-azure/pkg/providers/instancetype.ephemeralStorage(...)
        /home/uniemimu/go/pkg/mod/github.com/!azure/karpenter-provider-azure@v0.4.0/pkg/providers/instancetype/instancetype.go:310
github.com/Azure/karpenter-provider-azure/pkg/providers/instancetype.computeCapacity({0x24c0600, 0xc000bc1b60}, 0xc000eab618, 0xc000568070, 0xc000f3d290)
        /home/uniemimu/go/pkg/mod/github.com/!azure/karpenter-provider-azure@v0.4.0/pkg/providers/instancetype/instancetype.go:270 +0x1a2
github.com/Azure/karpenter-provider-azure/pkg/providers/instancetype.NewInstanceType({0x24c0600, 0xc000bc1b60}, 0xc000eab618, 0x0?, 0xc000568070, {0xc000014059?, 0xc0001347f8?}, {0xc000225500, 0x6, 0x8}, ...)
        /home/uniemimu/go/pkg/mod/github.com/!azure/karpenter-provider-azure@v0.4.0/pkg/providers/instancetype/instancetype.go:127 +0xdc
github.com/Azure/karpenter-provider-azure/pkg/providers/instancetype.(*Provider).List(0xc0011021e0, {0x24c0600, 0xc000bc1b60}, 0xc000568070, 0xc000f3d290)
        /home/uniemimu/go/pkg/mod/github.com/!azure/karpenter-provider-azure@v0.4.0/pkg/providers/instancetype/instancetypes.go:123 +0x6d5
github.com/Azure/karpenter-provider-azure/pkg/cloudprovider.(*CloudProvider).GetInstanceTypes(0xc000987ac0, {0x24c0600, 0xc000bc1b60}, 0x3718920?)
        /home/uniemimu/go/pkg/mod/github.com/!azure/karpenter-provider-azure@v0.4.0/pkg/cloudprovider/cloudprovider.go:153 +0x2c9
main.main()
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

--- a/pkg/providers/instancetype/instancetype.go
+++ b/pkg/providers/instancetype/instancetype.go
@@ -307,7 +307,7 @@ func memory(ctx context.Context, sku *skewer.SKU) *resource.Quantity {
 }
 
 func ephemeralStorage(nodeClass *v1alpha2.AKSNodeClass) *resource.Quantity {
-	return resource.NewScaledQuantity(int64(*nodeClass.Spec.OSDiskSizeGB), resource.Giga)
+	return resource.NewScaledQuantity(int64(lo.FromPtr(nodeClass.Spec.OSDiskSizeGB)), resource.Giga)
 }
 
 func pods(sku *skewer.SKU, kc *corev1beta1.KubeletConfiguration) *resource.Quantity {

